src/library.js

```diff
@@ -3696,6 +3696,36 @@ mergeInto(LibraryManager.library, {
     dbg('done preloading data files');
 #endif
   },
+
+  $handleAllocator__docs: '/** @constructor */',
+  $handleAllocator: function() {
+    this.allocated = [];
+    this.freelist = [];
+    this.get = function(id) {
+#if ASSERTIONS
+      assert(this.allocated[id] !== undefined);
+#endif
+      return this.allocated[id];
+    };
+    this.allocate = function(handle) {
+      let id;
+      if (this.freelist.length > 0) {
```

Comment:
@tlively .. for debugging purposes it might be nice to avoid re-using handles?  Or at least reuse them less.. we we can detect the use of a stale handle that happens to point to a newly allocated one.  WDYT?  Perhaps in debug builds we should just never reuse handles?

Reply:
Or perhaps we only start re-using handles after there are over N in the free list?   Something like 256 should be enough to ensure we don't reuse handles too eagerly?

Reply:
My gut feeling is that we shouldn't have the behavior of debugging and release builds diverge here. But if we would be ok with never reusing handles in debug mode, maybe we'd be ok with that in release mode, too? I'm not tied to the free list design as long as we have a common utility we can share. Maybe we can have a counter with an assertion that the counter doesn't get over 4 billion or whatever.

Reply:
For now I will leave this as is.